### PR TITLE
[CALCITE-6257] StarRocks dialect implementation

### PR DESCRIPTION
**Pull request by YiwenWu**
Implementing StarRocks dialect in Calcite
https://issues.apache.org/jira/browse/CALCITE-6257

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -524,10 +525,10 @@ private static String toSql(RelNode root, SqlDialect dialect,
 
   @Test void testSelectQueryWithHiveRollup() {
     String query = "select \"product_class_id\", \"product_id\", count(*) "
-            + "from \"product\" group by rollup(\"product_class_id\", \"product_id\")";
+        + "from \"product\" group by rollup(\"product_class_id\", \"product_id\")";
```
Comment (JiajunBernoulli) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:528]: Please avoid unnecessary higher values in order to preserve the original Git history.

**Comment by JiajunBernoulli** [comment]
LGTM

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 1 above)
Comment (YiwenWu) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: done, removed unnecessary changes

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 2]
```diff
@@ -7893,6 +7980,132 @@ private void checkLiteral2(String expression, String expected) {
...
+  }
+
+  @Test void testSelectQueryWithRollup() {
+    final String query = "select \"product_class_id\", \"product_id\", count(*) "
+        + "from \"product\" group by rollup(\"product_class_id\", \"product_id\")";
+    final String expectedStarRocks = "SELECT `product_class_id`, `product_id`, COUNT(*)\n"
+        + "FROM `foodmart`.`product`\n"
+        + "GROUP BY ROLLUP(`product_class_id`, `product_id`)";
+    sql(query).withStarRocks().ok(expectedStarRocks);
+    SqlDialect sqlDialect = sql(query).withHive().dialect;
```
Comment (macroguo-ghy) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:8105]: `withHive()` ?

**Review by macroguo-ghy — APPROVED** [review]
It looks good. I just have a comment about one test.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 2 above)
Comment (YiwenWu) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: done,  removed unnecessary changes.

**Pull request merged**